ssl/quic/quic_demux.c

```diff
@@ -544,7 +546,8 @@ static int demux_process_pending_urxe(QUIC_DEMUX *demux, QUIC_URXE *e)
         if (demux->default_cb != NULL) {
             /* Pass to default handler. */
             e->demux_state = URXE_DEMUX_STATE_ISSUED;
-            demux->default_cb(e, demux->default_cb_arg);
+            demux->default_cb(e, demux->default_cb_arg,
+                              dst_conn_id_ok ? &dst_conn_id : NULL);
```

Comment:
I don't see how? That is about calling with a pointer cast to a different function pointer type.

This is UB:

```c
struct z;
int f(const struct z *p);
typedef int (f_t)(struct z *p);

extern struct z *some_ptr;
f_t *fp = (f_t *)f;
fp(some_ptr);
```

This is not:

```c
struct z;
int f(const struct z *p);
typedef int (f_t)(const struct z *p);

extern struct z *some_ptr;
f_t *fp = (f_t *)f;
fp(some_ptr);
```

The type of `default_cb` is the same as when it is set by the setter for that field. No casts are performed anywhere here. So argument conversion will occur at call time in the normal way for any function. The issue arises when the real type of a function pointer is not known to the compiler because it has been forcibly cast to the wrong function pointer type. If the type of a function pointer known to a compiler is correct, of course it will perform argument conversion automatically.

Reply:
Oh, you're right, sorry.  Was looking at a function type that wasn't actually registered as a default_cb.